lib/minheap.js

```diff
@@ -38,26 +52,61 @@ class MinHeap {
   }
 
   // This method returns true if the heap is empty
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(1)
+  // Space complexity: O(1)
   isEmpty() {
-    throw new Error("Method not implemented yet...");
+    return this.store.length === 0
   }
 
   // This helper method takes an index and
   //  moves it up the heap, if it is less than it's parent node.
   //  It could be **very** helpful for the add method.
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(log n) - n: nodes in heap
+  // Space complexity: O(log n) or O(1)?
   heapUp(index) {
-    throw new Error("Method not implemented yet...");
+
+    let parentIndex = index % 2 === 0 ? (index - 2) / 2 : (index - 1) / 2;
+
+    while( index > 0 && (this.store[parentIndex].key > this.store[index].key)) {
+      this.swap(index, parentIndex);
+      index = parentIndex;
+      parentIndex = index % 2 === 0 ? (index - 2) / 2 : (index - 1) / 2;
+    }
   }
 
   // This helper method takes an index and 
   //  moves it up the heap if it's smaller
   //  than it's parent node.
+  // Time complexity: O(log n) - n: nodes in heap
+  // Space complexity: O(log n) or O(1)?
   heapDown(index) {
```

Comment:
For space complexity are you creating any new data structures?  Are you making recursive calls?  If not it's O(1).

Otherwise this works.